bindings/wasm/src/account/update/detach_method_relationship.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use js_sys::Promise;
+use std::rc::Rc;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_futures::future_to_promise;
+
+use identity::account::Account;
+use identity::account::DetachMethodRelationshipBuilder;
+use identity::account::IdentityUpdater;
+use identity::account::UpdateError::MissingRequiredField;
+use identity::core::OneOrMany;
+
+use identity::did::MethodRelationship;
+use wasm_bindgen::__rt::RefMut;
+use wasm_bindgen::__rt::WasmRefCell;
+
+use crate::account::wasm_account::WasmAccount;
+use crate::account::wasm_method_relationship::WasmMethodRelationship;
+use crate::error::Result;
+use crate::error::WasmResult;
+
+#[wasm_bindgen(js_class = Account)]
+impl WasmAccount {
+  /// Detaches the given relationship from the given method, if the method exists.
+  #[wasm_bindgen(js_name = detachMethodRelationships)]
+  pub fn detach_relationships(&mut self, options: &DetachMethodRelationshipOptions) -> Result<Promise> {
+    let relationships: Vec<MethodRelationship> = options
+      .relationships()
+      .into_serde::<OneOrMany<WasmMethodRelationship>>()
+      .map(OneOrMany::into_vec)
+      .wasm_result()?
+      .into_iter()
+      .map(Into::into)
+      .collect();
+
+    let account: Rc<WasmRefCell<Account>> = Rc::clone(&self.0);
+    let fragment: String = options
+      .fragment()
+      .ok_or(MissingRequiredField("fragment"))
+      .wasm_result()?;
+
+    let promise: Promise = future_to_promise(async move {
```

Comment:
Nit: only clone the `Rc` after input handling.

```suggestion
    let fragment: String = options
      .fragment()
      .ok_or(MissingRequiredField("fragment"))
      .wasm_result()?;

    let account: Rc<WasmRefCell<Account>> = Rc::clone(&self.0);
    let promise: Promise = future_to_promise(async move {
```